python/spatialdata-codec-writer/src/spatialdata_codec_writer/htj2k_encode.py

```
@@ -0,0 +1,205 @@
+        if node is None:
+            raise RuntimeError("Node.js is required for HTJ2K encode but was not found on PATH.")
+        if not _ENCODE_SCRIPT.is_file():
+            raise RuntimeError(f"HTJ2K encode script not found: {_ENCODE_SCRIPT}")
+        if not (openjph_vendor_dir() / "openjphjs.wasm").is_file():
+            raise RuntimeError(
+                f"Vendored OpenJPH WASM not found under {openjph_vendor_dir()}. "
+                "Run scripts/vendor-openjph-for-python.mjs before building the package."
+            )
+
+        self._lock = threading.Lock()
+        self._proc = subprocess.Popen(
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)